src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -53,8 +54,8 @@
      */
     public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
         this(syncOptions,
-            new TypeServiceImpl(syncOptions.getCtpClient()),
-            new CategoryServiceImpl(syncOptions.getCtpClient()));
+            new TypeServiceImpl(getRetryOn5xxClient(syncOptions.getCtpClient())),
```

Comment:
I wouldn't do it here on every time we need to get the client, but rather one time when it's injected and bound to the options.